src/NServiceBus.Core/Conventions/Conventions.cs

```diff
@@ -32,9 +42,22 @@ public bool IsMessageType(Type t)
                         {
                             return false;
                         }
-                        return IsMessageTypeAction(type) ||
-                               IsCommandTypeAction(type) ||
-                               IsEventTypeAction(type);
+
+                        foreach(var convention in conventions)
+                        {
+                            if(convention.IsMessageType(type)
+                            || convention.IsCommandType(type)
+                            || convention.IsEventType(type))
+                            {
+                                if(logger.IsDebugEnabled)
```

Comment:
Do we want to add more logs if the message satisfies more than just one type? e.g. it is both event and command?

Reply:
Each of these methods (is message, is command, is event) should be being called and logged separately from the outside. All this does is say "if a convention says it's a command or an event then it is also a message."

This could be confusing if a unit test calling `IsMessageType` returns false but then we're logging to say the convention matched. In short, I am :+1: to log when this happens.